src/routes/signin/webauthn.ts

```diff
@@ -0,0 +1,158 @@
+import { sendError } from '@/errors';
+import { ENV, getSignInResponse, getUserByEmail, gqlSdk } from '@/utils';
+import { RequestHandler } from 'express';
+
+import {
+  generateAuthenticationOptions,
+  verifyAuthenticationResponse,
+} from '@simplewebauthn/server';
+import { email, Joi } from '@/validation';
+
+export const signInWebauthnSchema = Joi.object({
+  email: email.required(),
+}).meta({ className: 'SignInWebauthnSchema' });
+
+export const signInVerifyWebauthnSchema = Joi.object({
+  email: email.required(),
+  credential: Joi.object().required(),
+}).meta({ className: 'SignInVerifyWebauthnSchema' });
+
+// A unique identifier for your website
+const rpID = 'localhost';
+// The URL at which registrations and authentications should occur
+const origin = `http://${rpID}:8000`;
+
+export const signInWebauthnHandler: RequestHandler<
+  {},
+  {},
+  {
+    email: string;
+  }
+> = async (req, res) => {
+  const { body } = req;
+  const { email } = body;
+
+  const user = await getUserByEmail(email);
+
+  if (!user) {
+    return sendError(res, 'invalid-email-password');
```

Comment:
probably better to use `user-not-found` instead

Reply:
Done ✅